modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtResultArtifactResolver.java

```diff
@@ -0,0 +1,154 @@
+        mapHomepage(pkg).ifPresent(a::addFact);
+
+        mapObservedLicense(pkg).ifPresent(a::addFact);
+        mapCopyrights(pkg).ifPresent(a::addFact);
+
+        return a;
+    }
+
+    private static Optional<ArtifactCoordinates> mapCoordinates(Package pkg) {
+        String namespace = pkg.getId().getNamespace();
+        String name = pkg.getId().getName();
+        String version = pkg.getId().getVersion();
+
+        switch (pkg.getId().getType().toLowerCase()) {
+            case "nuget":
+            case "dotnet":
+                return Optional.of(mapDotNetCoordinates(name, version));
+            case "maven":
+                return Optional.of(mapMavenCoordinates(namespace, name, version));
+            case "npm":
+                return Optional.of(mapJavaScriptCoordinates(name, version));
+            default:
+                return Optional.of(mapSimpleCoordinates(name, version));
+        }
+    }
+
+    private static ArtifactCoordinates mapDotNetCoordinates(String name, String version) {
+        DotNetCoordinates.DotNetCoordinatesBuilder c = new DotNetCoordinates.DotNetCoordinatesBuilder();
+        c.setPackageId(name);
+        c.setVersion(version);
+        return c.build();
+    }
+
+    private static ArtifactCoordinates mapMavenCoordinates(String namespace, String name, String version) {
+        MavenCoordinates.MavenCoordinatesBuilder c = new MavenCoordinates.MavenCoordinatesBuilder();
+        c.setGroupId(namespace);
+        c.setVersion(version);
+        c.setArtifactId(name);
+        return c.build();
+    }
+
+    private static ArtifactCoordinates mapJavaScriptCoordinates(String name, String version) {
+        JavaScriptCoordinates.JavaScriptCoordinatesBuilder c = new JavaScriptCoordinates.JavaScriptCoordinatesBuilder();
+        c.setName(name);
+        c.setVersion(version);
+        c.setArtifactId(name + "-" + version);
+        return c.build();
+    }
+
+    private static ArtifactCoordinates mapSimpleCoordinates(String name, String version) {
+        return new GenericArtifactCoordinates(name, version);
+    }
+
+    private static Optional<ArtifactSourceUrl> mapSourceUrl(Package pkg) {
+        final Stream<String> urls = Stream.of(pkg.getVcsProcessed().getUrl(), pkg.getSourceArtifact().getUrl());
+
+        return urls.filter(u -> !u.isEmpty())
+                .findFirst()
+                .map(ArtifactSourceUrl::new);
+    }
+
+    private static Optional<DeclaredLicenseInformation> mapDeclaredLicense(Package pkg) {
+        return Optional.of(pkg.getDeclaredLicenses())
+                .map(LicenseSupport::mapLicenses)
+                .map(DeclaredLicenseInformation::new);
+    }
+
+    private static Optional<ArtifactFilename> mapFilename(Package pkg) {
+        final Stream<RemoteArtifact> artifacts = Stream.of(pkg.getSourceArtifact(), pkg.getBinaryArtifact());
+
+        return artifacts
+                .filter(a -> !a.getUrl().isEmpty())
+                .findFirst()
```

Comment:
shouldn't there now be multiple ArtifactFilename entries possible in antenna?

Reply:
I don't know... maybe... probably :wink: Anyway, at this point I wanted to do a pure conversion of the original code. New features can be added later, now that it's easier to do that :grin:

Reply:
To my understanding, ORT distinguishes between binary and source artifact on a technology basis. I saw the result from Tuan and it showed in the Java case, that the binary artifact is the relevant thing, the source artifact is the packaged source jar, whereas in JavaScript, there is no binary artifact, i.e., the source artifact represents the real dependency used in the project. This code here looks like it takes randomly one of both, if both are present, that would not be the correct thing to do, or am I wrong?

Reply:
I've addressed this now in https://github.com/eclipse/antenna/pull/270.